pkg/k8s/pipelineRun.go

```diff
@@ -158,22 +159,27 @@ func (r *pipelineRun) UpdateState(state api.State) (*api.StateItem, error) {
 }
 
 // FinishState set end time stamp of the current (defined) state and add it to the history
+// if no current state is defined a new pickup state (A) with cretiontime of the pipelinerun as start time is created.
 // Returns the state details
 func (r *pipelineRun) FinishState() (*api.StateItem, error) {
 	state := r.cached.Status.StateDetails
-	if state.State != api.StateUndefined {
-		state.FinishedAt = metav1.Now()
-		his := r.cached.Status.StateHistory
-		his = append(his, state)
-		r.cached.Status.StateHistory = his
-		return &state, r.updateStatus()
+	now := metav1.Now()
+	if state.State == api.StateUndefined {
+		state.State = api.StatePickup
+		state.StartedAt = r.cached.ObjectMeta.CreationTimestamp
```

Comment:
Why is the start time of phase `pickup` the creation time of the resource object? It should be `now` (no-op phase with zero duration).

Reply:
The state is representing the time it took until the controller picks the object. Changed naming to new.